specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-12-01-preview/apimportalsettings.json

```diff
@@ -693,6 +693,390 @@
+          "PortalSettingsCsp"
+        ],
+        "operationId": "PortalSettingsCsp_Update",
+        "description": "Update the Content Security Policy (CSP) settings for the developer portal.",
+        "x-ms-examples": {
+          "ApiManagementPortalSettingsCspPatch": {
+            "$ref": "./examples/ApiManagementPortalSettingsCspPatch.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./definitions.json#/definitions/PortalSettingsCspContract"
+            },
+            "description": "Update the Content Security Policy (CSP) settings for the developer portal."
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/IfMatchRequiredParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "The settings were successfully updated."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PortalSettingsCsp"
+        ],
+        "operationId": "PortalSettingsCsp_CreateOrUpdate",
+        "description": "Create or update Content Security Policy (CSP) settings for the developer portal.",
+        "x-ms-examples": {
+          "PortalSettingsCspPut": {
+            "$ref": "./examples/ApiManagementPortalSettingsCspPut.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./definitions.json#/definitions/PortalSettingsCspContract"
+            },
+            "description": "Create or update parameters."
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/IfMatchOptionalParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```

Comment:
For any new PUT API we are asking devs to implement the api in such a way that it is a long-running operation.

Are you sure, this request is going to complete in  under 1 minute always.

Reply:
I'd say it's under a second. It's just settings.